### PR TITLE
Update R to 3.5.1

### PR DESCRIPTION
**Pull request by JanJecmen**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,11 +11,11 @@ set(R_INCLUDE_DIR ${R_HOME}/include)
...
 
 add_definitions(-g3)
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -std=c++11")
-set(CMAKE_CXX_FLAGS_DEBUG "-O0 -Wall -std=c++11 -DENABLE_SLOWASSERT")
-set(CMAKE_CXX_FLAGS "-Wall -std=c++11 -fno-rtti")
-set(CMAKE_C_FLAGS_RELEASE "-O2 -std=gnu99")
-set(CMAKE_C_FLAGS_DEBUG "-Og -std=gnu99 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS "-Wall -std=c++17 -fno-rtti")
```
Comment (mhyee) [inline CMakeLists.txt:16]: Well, if we're skipping R 3.4, might as well skip C++14!

I look forward to ~spamming~ using:

- generic lambdas
- s suffix for `std::string`
- `static_assert` without the message string
- nested namespace definitions
- static `if`
- `std::optional`

**Review by mhyee — COMMENTED** [review]
Another thing to update is the Travis config. It's set to Trusty (14.04!), so of course GCC is version 4.8.4 and doesn't understand `-std=c++17`. The latest Ubuntu LTS is Bionic Beaver (18.04).

**Comment by JanJecmen** [comment]
Oh you fixed it! Thanks!!

**Comment by JanJecmen** [comment]
TBR then, I guess :)

**Comment by o-** [comment]
Yet another :tada:

**Pull request merged**